libursa/OnDiskDataset.cpp

```diff
@@ -245,22 +246,26 @@ void OnDiskDataset::merge(const fs::path &db_base, const std::string &outname,
         task->work_estimated = NUM_TRIGRAMS * index_types.size();
     }
 
+    spdlog::info("Pre-checks succeeded, merge can begin.");
+
     std::set<std::string> index_names;
-    for (IndexType index_type : index_types) {
-        std::string index_name =
-            get_index_type_name(index_type) + "." + outname;
+    for (IndexType ndxtype : index_types) {
```

Comment:
```suggestion
    for (const auto& ndxtype : index_types) {
```

Tho it requires `get_index_type_name` to be marked as `const` (it is) and `OnDiskIndex::on_disk_merge` to take `const IndexType merge_type`.